lib/Unix/PID/Tiny.pm

```diff
@@ -265,6 +267,8 @@ sub pid_file_no_unlink {
 
     syswrite( $pid_fh, int( abs($newpid) ) );
 
+    rename( $tmp_pid_file, $pid_file );
```

Comment:
oh yeah, almost forgot. `rename()` doesn’t work on open file handles on some systems I think we'd have to close it first, rename, then reopen it to get a new handle to make it more portable.